Jenkinsfile

```diff
@@ -27,34 +30,43 @@ pipeline {
+                        // Iterate over each plugin directory
+                        pluginDirectories.split().each { repo ->
+                            // Truncate hanging "/" for each directory
+                            def pluginName = repo.getAt(0..(repo.length() - 2))
+                            // Check if VERSION file for each plugin file has changed
+                            def isChanged = "0"
 
-                        if (env.GIT_PREVIOUS_SUCCESSFUL_COMMIT) {
-                            isChanged = """${sh (
-                                script: "git diff --name-only ${GIT_PREVIOUS_SUCCESSFUL_COMMIT} ${GIT_COMMIT} | grep ${pluginName}/VERSION",
-                                returnStatus: true
-                            )}"""
-                        }
-                        if (isChanged == "0" && pluginName != "utils") {
-                            dir("${WORKSPACE}/${pluginName}") {
-                                def dockerVersion = readFile(file: 'VERSION').trim()
-                                docker.withRegistry('https://registry-1.docker.io/v2/', 'f16c74f9-0a60-4882-b6fd-bec3b0136b84') {
-                                    def image = docker.build("labshare/${pluginName}", '--no-cache ./')
-                                    image.push()
-                                    image.push(dockerVersion)
+                            if (env.GIT_PREVIOUS_SUCCESSFUL_COMMIT) {
+                                isChanged = """${sh (
+                                    script: "git diff --name-only ${GIT_PREVIOUS_SUCCESSFUL_COMMIT} ${GIT_COMMIT} | grep ${pluginName}/VERSION",
+                                    returnStatus: true
+                                )}"""
+                            }
+                            if (isChanged == "0" && pluginName != "utils") {
+                                dir("${WORKSPACE}/${pluginName}") {
+                                    def dockerVersion = readFile(file: 'VERSION').trim()
+                                    docker.withRegistry('https://registry-1.docker.io/v2/', 'f16c74f9-0a60-4882-b6fd-bec3b0136b84') {
+                                        def image = docker.build("labshare/${pluginName}", '--no-cache ./')
+                                        image.push()
+                                        image.push(dockerVersion)
+                                    }
+                                    
+                                    env.PROJECT_NAME = "${pluginName}"
+                                    env.FULL_DESC = readFile(file: 'README.md')
+                                    env.BRIEF_DESC = "${PROJECT_URL}/tree/master/${PROJECT_NAME}"
+                                }
+                                withCredentials([usernamePassword(credentialsId: 'f16c74f9-0a60-4882-b6fd-bec3b0136b84', usernameVariable: 'DOCKER_USER', passwordVariable: 'DOCKER_PW')]) {
+                                    sh "sh ./update.sh"
```

Comment:
Why do you need `sh "sh ...`? Would `sh "./update.sh" work?

Reply:
If the file doesn't have execute permissions (chmod +x), you can't just run it as an executable. The first `sh` is just the Jenkins agent syntax that opens a shell to run the command in.